python/tvm/relay/frontend/onnx.py

```diff
@@ -678,25 +678,32 @@ def _impl_v1(cls, inputs, attr, params):
         # When performing a batch matmul, we need to properly handle N-dim shapes.
         if a_rank > 2 or b_rank > 2:
 
-            def flatten_to_3d(x, x_shape):
+            def flatten_to_nd(x, x_shape, nd=3):
```

Comment:
Can we have a shortcut to return `x` directly when it is already `nd` so that we can void the unnecessary reshape?

Reply:
Adopted, good suggestion!